semver/gtr.ts

```diff
@@ -1,12 +1,11 @@
 // Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
 import type { Range, SemVer, SemVerRange } from "./types.ts";
-import { rangeMax } from "./range_max.ts";
-import { gt } from "./gt.ts";
+import { greaterThanRange } from "./greater_than_range.ts";
 
-/** Checks to see if the version is greater than all possible versions of the range. */
-export function gtr(
-  version: SemVer,
-  range: SemVerRange | Range,
-): boolean {
-  return gt(version, rangeMax(range));
+/**
+ * Checks to see if the version is greater than all possible versions of the range.
+ * @deprecated (will be removed after 0.215.0) Use {@linkcode greaterThanRange} instead.
+ */
+export function gtr(version: SemVer, range: SemVerRange | Range): boolean {
+  return greaterThanRange(range, version);
```

Comment:
Can you please use the new [`warnOnDeprecatedApi()`](https://github.com/denoland/deno_std/blob/0a32c8f5e82f939e8a46f8c027c1fb5b3040590a/internal/warn_on_deprecated_api.ts#L33)? Ditto with other deprecated functions.

Reply:
I'll have a look

Reply:
I like that new API, but this seems like the addition of this should be automated to stay in sync with`@deprecated` tags.

Reply:
I agree, and I think we should explore that idea in a separate issue. But for now, can we please use `warnOnDeprecatedApi()`?

Reply:
done